stan/math/prim/fun/factor_cov_matrix.hpp

```diff
@@ -18,35 +18,36 @@ namespace math {
  * @tparam T type of elements in the matrix and arrays
  * @param[in] Sigma covariance matrix
  * @param[out] CPCs fill this unbounded (does not resize)
- * @param[out] sds fill this unbounded (does not resize)
+ * @param[out] sds fill this unbounded
  * @return false if any of the diagonals of Sigma are 0
  */
-template <typename T>
-bool factor_cov_matrix(
-    const Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>& Sigma,
-    Eigen::Array<T, Eigen::Dynamic, 1>& CPCs,
-    Eigen::Array<T, Eigen::Dynamic, 1>& sds) {
+template <typename T_Sigma, typename T_CPCs, typename T_sds,
+          require_eigen_t<T_Sigma>* = nullptr,
+          require_all_eigen_vector_t<T_CPCs, T_sds>* = nullptr,
+          require_all_same_vt<T_Sigma, T_CPCs, T_sds>* = nullptr>
+bool factor_cov_matrix(const T_Sigma& Sigma, T_CPCs& CPCs, T_sds& sds) {
```

Comment:
Side Note: Another thing I goofed / am goofing with is removing CPCs and sds here as in/out parameters

https://github.com/stan-dev/math/blob/refactor/cov-corr-constrain/stan/math/prim/fun/factor_cov_matrix.hpp

Reply:
I prefere in/out parameters to returning tuples. I made some more changes so that these two need not be copied to x after this function returns. Instead this function gets segments of that x.